torch/_dynamo/utils.py

```diff
@@ -3644,10 +3644,11 @@ def build_checkpoint_variable(**options):
 def is_compile_supported(device_type):
     from .eval_frame import is_dynamo_supported
 
+    type = torch.device(device_type).type
     compile_supported = is_dynamo_supported()
-    if device_type == "cpu":
+    if type == "cpu":
         pass
-    elif device_type in ["cuda", "xpu"] and compile_supported:
+    elif type in ["cuda", "xpu"] and compile_supported:
```

Comment:
@albanD @EikanWang Please take a look at here thanks!

@FFFrog For out-of-tree backends, we should provide the ability to check whether they support `torch.compile`, right?

Reply:
Of course, one possible implementation is as follows (note the exception routine):

```Python
module = getattr(torch, torch._C._get_privateuse1_backend_name())
module.is_compile_supported()
```